src/hooks/useLocation.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback, useEffect, useState } from '@redneckz/uni-jsx/lib/hooks';
+import { useAsyncData } from '@redneckz/uni-jsx/lib/hooks/useAsyncData';
+import { fetchJSONUnsafe } from '../utils/fetchJSON';
+
+const REGION_URL = '/api/v1/region';
+
+export interface RegionData {
+  id?: number;
+  code?: string;
+  name?: string;
+}
+
+interface LocationStorageData {
+  manual?: boolean;
+  region: RegionData;
+}
+
+const localStorageLocation = globalThis?.localStorage.getItem('location');
+const currentLocation: LocationStorageData | null = localStorageLocation
+  ? JSON.parse(localStorageLocation)
+  : null;
+
+export function useLocation(
+  defaultLocation: string,
+): [RegionData, (newLocation: RegionData) => void] {
+  const [location, setLocation] = useState<RegionData>({ name: defaultLocation });
+  const { data } = useAsyncData<RegionData>(REGION_URL, fetchJSONUnsafe);
+
+  useEffect(() => {
+    setLocation(
+      currentLocation && currentLocation.manual
```

Comment:
```suggestion
      currentLocation?.region && currentLocation?.manual
```

Возможно что стоит добавить проверку и на наличие `region`, чтобы уж наверняка